Gemfile

```diff
@@ -30,7 +30,14 @@ gem 'redcarpet', '~> 3.4' if RUBY_PLATFORM != 'java'
 gem 'rspec', '~> 3.10'
 gem 'rspec-collection_matchers', '~> 1.1'
 gem 'rspec_junit_formatter', '>= 0.4.1'
-gem 'rspec_n', '~> 1.3' if RUBY_VERSION >= '2.3.0'
+if RUBY_VERSION >= '2.3.0'
+  gem 'rspec_n', '~> 1.3'
+
+  # Ancient bundler mistakenly installs incompatible cri version for Ruby 2.3.
+  # We can't update bundler because it would mean going from bundler v1 to v2,
+  # and that breaks many gems that we test with in Ruby 2.3.
+  gem 'cri', '< 2.15.11' if RUBY_VERSION < '2.4.0'
```

Comment:
Minor: I think we can just live with `rspec_n` being only available on >= Ruby 2.4, to simplify things a bit here.